src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -5,114 +5,283 @@
+                a.setDueDate(dto.dueDate());
+                assignmentRepository.save(a);
+                return new AssignmentDTO(
+                    a.getAssignmentId(),
+                    a.getTitle(),
+                    a.getDueDate(),
+                    a.getSection().getCourse().getCourseId(),
+                    a.getSection().getSecId(),
+                    a.getSection().getSectionNo()
+                );
+            }
+        }
 
-        return null;
-    }
+        // delete assignment for a section
+        // logged in user must be instructor of the section
+        @DeleteMapping("/assignments/{assignmentId}")
+        @PreAuthorize("hasAuthority('SCOPE_ROLE_INSTRUCTOR')")
+        public void deleteAssignment ( @PathVariable("assignmentId") int assignmentId, Principal principal) {
+            String instructorEmail = principal.getName();
+            Assignment a = assignmentRepository.findById(assignmentId).orElse(null);
 
-    // instructor uploads grades for assignment
-    // user must be instructor for the section
-    @PutMapping("/grades")
-    public void updateGrades(@RequestBody List<GradeDTO> dlist) {
+            List<Grade> grade = gradeRepository.findByAssignmentId(assignmentId);
 
-        // TODO
+            if (a != null) {
+                Section section = a.getSection();
+                if (section.getInstructorEmail() == null || !section.getInstructorEmail().equals(instructorEmail)) {
+                    throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "unauthorized");
+                }
 
-        // for each grade in the GradeDTO list, retrieve the grade entity
-        // update the score and save the entity
+                for (Grade g : grade) {
+                    if (g.getScore() != null){
+                        throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "cannot delete an assignment if a grade has been recorded");
+                    }
+                    /**gradeRepository.delete(g);***/
+                }
+                assignmentRepository.delete(a);
+            }
+        }
 
-    }
+        // instructor gets grades for assignment ordered by student name
+        // user must be instructor for the section
+        @GetMapping("/assignments/{assignmentId}/grades")
+        @PreAuthorize("hasAuthority('SCOPE_ROLE_STUDENT')")
+        public List<GradeDTO> getAssignmentGrades ( @PathVariable("assignmentId") int assignmentId,
+                                                    Principal principal){
 
+//        int sectionNo = assignmentRepository.findSectionNoByAssignmentId(assignmentId);
+            Assignment a = assignmentRepository.findById(assignmentId).orElse(null);
+            if(a==null){
+                throw  new ResponseStatusException( HttpStatus.NOT_FOUND, "not found");
+            }
 
+            List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(a.getSection().getSectionNo());
```

Comment:
Need to check that the user calling the method is the user of the sections. So you can use ```principal``` argument to get instructor email name. Then you can check that instructor email name matches the assignment. Example ```a.getSection().getInstructorEmail()```. Check ```getAssignments``` method at top of the file for an example of this.

Reply:
I just pushed updates in that methods, could you please check if it looks good?